classes/patterns/class-homepage.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * HomePage pattern class.
+ *
+ * @package P4GBKS
+ * @since 0.1
+ */
+
+namespace P4GBKS\Patterns;
+
+/**
+ * This class is used for returning a home page template.
+ *
+ * @package P4GBKS\Patterns
+ */
+class HomePage extends Block_Pattern {
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function get_name(): string {
+		return 'p4/home-page-pattern-layout';
+	}
+
+	/**
+	 * @inheritDoc
+	 *
+	 * @param array $params Optional array of parameters for the config.
+	 */
+	public static function get_config( $params = [] ): array {
+		return [
+			'title'      => __( 'Homepage', 'planet4-blocks-backend' ),
+			'blockTypes' => [ 'core/post-content' ],
+			'categories' => [ 'layouts' ],
+			'content'    => '
+			<!-- wp:group -->
+			<div class="wp-block-group">
+
+				<!-- wp:planet4-blocks/carousel-header -->
+				<div class="wp-block-planet4-blocks-carousel-header">
+					<div data-hydrate="planet4-blocks/carousel-header" data-attributes="{&quot;carousel_autoplay&quot;:false,&quot;slides&quot;:[{&quot;image&quot;:null,&quot;focal_points&quot;:{},&quot;header&quot;:&quot;&quot;,&quot;description&quot;:&quot;&quot;,&quot;link_text&quot;:&quot;&quot;,&quot;link_url&quot;:&quot;&quot;,&quot;link_url_new_tab&quot;:false}]}" data-reactroot="">
+						<section class="block block-header alignfull carousel-header ">
+							<div class="carousel-wrapper-header">
+								<div class="carousel-inner" role="listbox">
+									<div class="carousel-item active">
+										<div class="carousel-item-mask">
+											<div class="background-holder">
+												<img style="object-position:NaN% NaN%"/>
+											</div>
+											<div class="carousel-caption">
+												<div class="caption-overlay"></div>
+												<div class="container main-header">
+													<div class="carousel-captions-wrapper">
+														<h2></h2>
+														<p></p>
+													</div>
+												</div>
+											</div>
+										</div>
+									</div>
+								</div>
+							</div>
+						</section>
+					</div>
+				</div>
+				<!-- /wp:planet4-blocks/carousel-header -->
+
+				' . Issues::get_config()['content'] . '
+
+				<!-- wp:spacer {"height":"88px"} -->
+				<div style="height:88px" aria-hidden="true" class="wp-block-spacer"></div>
+				<!-- /wp:spacer -->
+
+				<!-- wp:planet4-blocks/articles {
+					"article_heading":"' . __( 'Read our Stories', 'planet4-blocks' ) . '"
+				} /-->
+
+				<!-- wp:spacer {"height":"56px"} -->
+				<div style="height:56px" aria-hidden="true" class="wp-block-spacer"></div>
+				<!-- /wp:spacer -->
+
+				' . SideImageWithTextAndCta::get_config( [] )['content'] . '
+
+				<!-- wp:spacer {"height":"30px"} -->
+				<div style="height:30px" aria-hidden="true" class="wp-block-spacer"></div>
+				<!-- /wp:spacer -->
+
+				' . SideImageWithTextAndCta::get_config( [ 'media_position' => 'right' ] )['content'] . '
+
+				<!-- wp:spacer {"height":"56px"} -->
```

Comment:
Here you marked `56px`, but on the following line it's `54px`, so I get an error in the editor when adding the layout

Reply:
Good catch, thanks!
That happens because I've updated the code directly to the demo page instead of creating a new layout. And after the UAT is passed then I set all these values on my local file.